packages/hub/routes/merchant-infos.ts

```diff
@@ -64,6 +72,50 @@ export default class MerchantInfosRoute {
     ctx.body = serialized;
     ctx.type = 'application/vnd.api+json';
   }
+
+  async getValidation(ctx: Koa.Context) {
+    let slug = ctx.query?.slug;
+    let isValidSlug = await ensureValidSlug(ctx, slug, this.merchantInfoQueries);
+
+    if (!isValidSlug) {
+      return;
+    }
+
+    ctx.status = 200;
+    ctx.body = {
+      slugAvailable: true,
+      info: 'Merchant slug is available',
+    };
+    ctx.type = 'application/vnd.api+json';
+  }
+}
+
+async function ensureValidSlug(ctx: Koa.Context, slug?: string | string[], queries?: MerchantInfoQueries) {
+  let detail = '';
+
+  if (!slug) {
+    detail = `Slug cannot be undefined`;
+  } else if (typeof slug === 'object') {
+    detail = `Slug cannot be an array`;
+  } else if (validateMerchantId(slug)) {
```

Comment:
`validateMerchantId(slug)` is called twice in this case. Maybe better to assign it to a variable first and then check